qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1413,6 +1413,18 @@ def set_html(self, html, base_url=QUrl()):
         # percent encoded content is 2 megabytes minus 30 bytes.
         self._widget.setHtml(html, base_url)
 
+    def set_lifecycle_state(
+            self,
+            new_state: QWebEnginePage.LifecycleState
+    ) -> None:
+        """Set the lifecycle state of the current tab."""
+        if sip.isdeleted(self._widget):
+            log.webview.debug("Ignoring page lifecycle update for deleted widget")
```

Comment:
Is this check needed at all? If so, should it maybe be in `_on_recommended_state_changed` instead? I'd be vary of silently ignoring calls if some other part in the code calls it explicitly, I'd rather fail loudly than silently in that case.

Reply:
I will test-drive it with the `@pyqtSlot` change and see! It's been a while so I don't remember why it's there.